tests/foreman/cli/test_ansible.py

```diff
@@ -677,3 +679,273 @@ def _finalize():
+    def test_positive_ansible_dynamic_inventory(
+        self,
+        request,
+        target_sat,
+        module_org,
+        module_location,
+        module_ak_with_cv,
+        aap_client,
+        rhel_contenthost,
+        auth_type,
+    ):
+        """Verify AAP is able to pull a dynamic inventory of hosts from Satellite,
+        with admin and non-admin user.
+
+        :id: ad25aee5-4ea3-4743-a301-1c6271856g19
+
+        :steps:
+            1. Register content host to Satellite with admin and non-admin user
+            2. Update Satellite type credentials in AAP
+            3. Find Satellite inventory, inventory source and sync inventory source in AAP
+            4. Verify registered hosts are added to Satellite inventory
+
+        :expectedresults: All hosts managed by Satellite are added to Satellite inventory.
+        """
+        creds_name = settings.AAP_INTEGRATION.satellite_credentials
+        inventory_name = settings.AAP_INTEGRATION.satellite_inventory
+
+        password = settings.server.admin_password
+        if auth_type == 'admin':
+            login = gen_string('alpha')
+            user = target_sat.api.User(
+                admin=True,
+                login=login,
+                password=password,
+                organization=[module_org],
+                location=[module_location],
+            ).create()
+        else:
+            login = gen_string('alpha')
+            role = target_sat.api.Role(
+                name=gen_string('alpha'), location=[module_location], organization=[module_org]
+            ).create()
+            for perm in ['view_hosts', 'view_hostgroups', 'view_facts']:
+                permission = target_sat.api.Permission().search(query={'search': f'name={perm}'})
+                target_sat.api.Filter(permission=permission, role=role).create()
+            user = target_sat.api.User(
+                role=[role],
+                admin=False,
+                login=login,
+                password=password,
+                organization=[module_org],
+                location=[module_location],
+            ).create()
+            target_sat.cli.User.add_role(
+                {'login': user.login, 'role': 'Ansible Tower Inventory Reader'}
+            )
+        result = rhel_contenthost.register(
+            module_org,
+            module_location,
+            module_ak_with_cv.name,
+            target_sat,
+            auth_username=user.login,
+            auth_password=password,
+            force=True,
+        )
+        assert result.status == 0, f'Failed to register host: {result.stderr}'
+
+        if is_open('SAT-28613') and auth_type == 'non-admin':
+            host = rhel_contenthost.nailgun_host
+            host.location = module_location
+            host.update(['location'])
+
+        # Find the Satellite credentials in AAP and update it for target_sat.hostname and user credentials
+        creds_list = aap_client.get(
```

Comment:
This one looks like another candidate for a fixture. I see you use the same code in the other test too.